app/Http/Controllers/UserController.php

```diff
@@ -143,4 +156,36 @@ public function check(Request $request)
         }
         return view('users.check',compact('score','question','answers','choices','collection','collection1','categoryId','lessonId'));
     }
+
+    public function wordsLearned()
+    {
+        $userId = auth()->user()->id;
+        $name = auth()->user()->name;
+        $answers = User::find($userId)->answer()->get();
+
+        $wordsLearned = 0;
+        $currentUserId = auth()->user()->id;
+        $results= Result::where('user_id','=',$currentUserId)->get();
+        for($i=0;$i<$results->count();$i++){
+            $wordsLearned = $wordsLearned+ $results[$i]->score;
+        }
+
+        $answerCollection = collect();
+        for($i=0;$i<$answers->count();$i++){
+            $answerCollection->push($answers[$i]->choice_id);
+        }
+
+        $correctAnswers = collect();
```

Comment:
Try to search collection mapping in laravel it can be helpful to filter what data is just needed. calling `collect` function is redundant